src/xtopcom/xvm/xsystem_contracts/src/xtable_reward_claiming_contract.cpp

```diff
@@ -95,8 +97,151 @@ void xtop_table_reward_claiming_contract::update_vote_reward_record(common::xacc
+                                                                  std::map<std::string, std::map<std::string, std::string>> & pledge_votes_map,
+                                                                  std::map<std::string, uint64_t> & stored_expire_token_map) {
+    { MAP_COPY_GET(key, voters, data::xdatautil::serialize_owner_str(sys_contract_sharding_vote_addr, table_id)); }
+    xinfo("[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming table_id:%d,vote_valid_rate vote maps %s size: %d, pid: %d",
+          table_id,
+          key.c_str(),
+          voters.size(),
+          getpid());
+
+    for (auto const & entity : voters) {
+        auto const & account = entity.first;
+        std::map<std::string, std::string> pledge_votes;
+        MAP_COPY_GET(data::XPROPERTY_PLEDGE_VOTE_KEY, pledge_votes, account);
+        if (pledge_votes.empty()) {
+            xdbg("[xtop_table_reward_claiming_contract::recv_voter_dividend_reward] reward-claiming pledge votes is empty,table_id:%d,account:%s", table_id, account);
+            // do nothing
+            continue;
+        }
+        pledge_votes_map.insert({account, pledge_votes});
+
+        std::string val = STRING_GET2(data::XPROPERTY_EXPIRE_VOTE_TOKEN_KEY, account);
+        stored_expire_token_map.insert({account, xstring_utl::touint64(val)});
+    }
+}
+
+void xtop_table_reward_claiming_contract::calc_section_votes_table_and_adv_vote(std::map<std::string, std::string> & voters,
+                                                                                std::map<std::string, std::map<std::string, std::string>> & pledge_votes_map,
+                                                                                std::map<std::string, uint64_t> & stored_expire_token_map,
+                                                                                std::map<std::string, std::map<std::string, uint64_t>> & votes_table_map,
+                                                                                std::map<std::string, std::string> & adv_votes,
+                                                                                uint64_t table_id,
+                                                                                common::xlogic_time_t m_timer_height) {
+    for (auto const & entity : voters) {
+        auto const & account = entity.first;
+        auto const & vote_table_str = entity.second;
+        base::xstream_t stream(base::xcontext_t::instance(), (uint8_t *)vote_table_str.c_str(), (uint32_t)vote_table_str.size());
+        std::map<std::string, uint64_t> votes_table;
+        stream >> votes_table;
+        pledge_votes_map.find(account);
+        std::map<std::string, std::string> pledge_votes;
+
+        auto iter = pledge_votes_map.find(account);
+        if (iter == pledge_votes_map.end()) {
+            continue;
+        }
+        pledge_votes = iter->second;
```

Comment:
it looks like pledge_votes never changed? if so, `auto const & pledge_votes = top::get<std::map<std::string, std::string> const>(*iter);